go/buildhelper/node.go

```diff
@@ -0,0 +1,207 @@
+package buildhelper
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"fmt"
+	"math/big"
+	"strings"
+	"sync"
+
+	"github.com/ethereum/go-ethereum"
+
+	"github.com/obscuronet/obscuro-playground/go/buildhelper/buildconstants"
+	"github.com/obscuronet/obscuro-playground/go/buildhelper/helpertypes"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/obscuronet/obscuro-playground/go/log"
+	"github.com/obscuronet/obscuro-playground/go/obscurocommon"
+	"github.com/obscuronet/obscuro-playground/go/obscuronode/nodecommon"
+)
+
+var (
+	nonceLock sync.RWMutex
+)
+
+type EthNode struct {
+	port      uint
+	ipaddress string
+	apiClient *EthAPI
+	id        common.Address
+}
+
+func NewEthNode(id common.Address, ipaddress string, port uint) (obscurocommon.L1Node, error) {
+	apiClient := NewEthAPI(ipaddress, port)
+	err := apiClient.Connect()
+	if err != nil {
+		return nil, fmt.Errorf("unable to connect to the eth node - %w", err)
+	}
+
+	log.Log(fmt.Sprintf("Initializing eth node at contract: %s", buildconstants.CONTRACT_ADDRESS))
+	return &EthNode{
+		ipaddress: ipaddress,
+		port:      port,
+		apiClient: apiClient,
+		id:        id,
+	}, nil
+}
+
+func (e *EthNode) RPCBlockchainFeed() []*types.Block {
+	var availBlocks []*types.Block
+
+	block, err := e.apiClient.ApiClient.BlockByNumber(context.Background(), nil)
+	if err != nil {
+		panic(err)
+	}
+	availBlocks = append(availBlocks, block)
+
+	for {
+		parentHash := block.ParentHash()
+		// todo set this to genesis hash
+		if parentHash.Hex() == "0x0000000000000000000000000000000000000000000000000000000000000000" {
+			break
+		}
+
+		block, err = e.apiClient.ApiClient.BlockByHash(context.Background(), block.ParentHash())
+		if err != nil {
+			fmt.Printf("ERROR %v\n", err)
+		}
+
+		availBlocks = append(availBlocks, block)
+	}
+
+	// todo double check the list is ordered [genesis, 1, 2, 3, 4, ..., last]
+	for i, j := 0, len(availBlocks)-1; i < j; i, j = i+1, j-1 {
+		availBlocks[i], availBlocks[j] = availBlocks[j], availBlocks[i]
+	}
+	return availBlocks
+}
+
+func (e *EthNode) BroadcastTx(t obscurocommon.EncodedL1Tx) {
```

Comment:
can you add a comment on this that this conversion logic is temporary?
basically, there should be a follow-up task ( would be good to raise it) to make sure that we remove the dependence on "L1TxType" and use proper ethereum transaction.

Reply:
Yeah, I actually think it's ok to encapsulate these. But it might be a problem with the enclave blockchain validation.

Let me raise that task nonetheless 👍